sample_information_validator.py

```diff
@@ -51,44 +60,97 @@ def validate(self, cell):
-                logger.warning(
+                logger.error(
                     'Cell {} is numeric but empty'.format(cell.coordinate)
                 )
+                return False
             else:
                 raise
 
+    def validate_vol(self,cell): # TODO different volumes for different input types! requires changes in data sheet
+        """Checks entry for volume"""
+        if cell.value is None:
+            logger.error(
+                'No sample volume given in cell {}'.format(cell.coordinate)
+            )
+            return False
+        else:
+            return True
+
+    def validate_rin(self,cell, sample_type):
+        """Checks entry for RIN in RNA samples only"""
+        if cell.value is None:
+            logger.error(
+                'No RIN value given in cell {}'.format(cell.coordinate)
+            )
+            return False
+        elif cell.value <8:
+            logger.warning(
+                'RIN value in cell {} is below recommendation'.format(cell.coordinate)
+            )
+            return True
+        else:
+            return True
 
-def validate_column(sheet, column_letter, row_start,
-                    row_end, validator, validator_attr={}):
-    """Validates a section of a column
+def validate_column(sheet, plate_id, concentration_letter, volume_letter, rin_letter, cell_sample_type, sample_rowID,
+                    validator, validator_attr={}):
+    """Validates all rows with a sample ID
 
-    Given the column letter and which rows to validate:
-    Initiates the given validator with the optional attributes
+    First checks for existence of a plate ID and if user changed the default.
+    Then, given the column letter and which rows to validate:
+    Initiates the given validators for concentration, volumne and RIN (RNA samples only) with the optional attributes
     Loops through all the given cells
     validates them individually.
     """
-
+    if(sheet[plate_id].value == None or sheet[plate_id].value == 'P####P#'):
+        logger.error(
+            'Missing PLATE ID'
+            )
     validator = NumericValidator(**validator_attr)
     passes = 0
     total = 0
-    for row_nr in range(row_start, row_end):
+    for row_nr in sample_rowID:
         total += 1
-        cell_id = "{col}{row_nr}".format(col=column_letter, row_nr=row_nr)
-        result = validator.validate(sheet[cell_id])
-        if result:  # Test passed
-            passes += 1
+        cell_id_conc = "{col}{row_nr}".format(col=concentration_letter, row_nr=row_nr)
+        cell_id_vol = "{col}{row_nr}".format(col=volume_letter, row_nr=row_nr)
+        cell_id_rin = "{col}{row_nr}".format(col=rin_letter, row_nr=row_nr)
+        result_conc = validator.validate_conc(sheet[cell_id_conc])
+        result_vol = validator.validate_vol(sheet[cell_id_vol])
+        if (sheet[cell_sample_type].value == 'Total RNA' or sheet[cell_sample_type].value == "mRNA" or sheet[cell_sample_type].value == 'Small RNA'):
```

Comment:
I'd wrap this line into two or more rows. And also perhaps having the brackets around each or-clause and not around the whole thing. Functionally it doesn't matter, it's just for readability.